source/state_representation/include/state_representation/space/cartesian/CartesianPose.hpp

```diff
@@ -225,6 +247,12 @@ class CartesianPose : public CartesianState {
    */
   std::vector<double> norms(const CartesianStateVariable& state_variable_type = CartesianStateVariable::POSE) const override;
 
+  /**
+   * @brief Compute the inverse of the current CartesianPose
+   * @return the inverse corresponding to b_S_f (assuming this is f_S_b)
+   */
+  CartesianPose inverse() const;
```

Comment:
The `CartesianState inverse() const` is not deleted in this header, which means this new declaration is hiding the base method. It would be safer to delete the base inverse method in this header as well.

Reply:
If I add the line
```
  CartesianState inverse() const = delete;
```
to the CartesianPose header, I cannot build state_representation anymore

Reply:
Ah ok, my bad. Then instead you should mark the CartesianState::inverse as virtual and explicitly override the derived version here.

```c++
// in CartesianState.hpp
virtual CartesianState CartesianState::inverse() const;

// in CartesianPose.hpp
CartesianPose CartesianPose::inverse() const override;
```

This should work because the return type of the derived function is a co-variant (the derived class) of the base function (the base class).

Reply:
No, that doesn't work unfortunately...

Reply:
Actually, there are other functions like that (the `copy` one) that are not defined virtual or deleted but that hide the CartesianState function